library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/server/auth/DefaultGafferPopAuthenticator.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.tinkerpop.server.auth;
+
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticatedUser;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticationException;
+import org.apache.tinkerpop.gremlin.server.auth.Authenticator;
+
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The default authenticator class for GafferPop, this should not
+ * be used in production as it allows all user and password combinations.
+ * The class is intended as a template for an instance specific class
+ * that hooks into a proper authorisation mechanism such as LDAP etc.
+ */
+public class DefaultGafferPopAuthenticator implements Authenticator {
```

Comment:
I'm wondering if this should be called "default" if we don't want it to be used except for testing.

Reply:
That is probably true maybe could be changed to `ExampleGafferPopAuthenticator` as its not really a production ready class